docs/deployment.rst

```diff
@@ -0,0 +1,36 @@
+Safeguard
+---------
+
+To balance contention between topics, certain safeguard-limits are placed on each topic-partition pull.
```

Comment:
"To balance contention between topics...". I think we should first explain the contention, how you can run into it, why etc.
"certain safeguard-limits". What sort of limits? Also let's drop the work safeguard-